Feature/nieuws

PerryWdm opened the pull request:
# Definition of Done

## Functionele definities 

- [x] De acceptatiecriteria van de user story zijn gehaald. 

- [] Elk stuk werk is nagekeken door minimaal 2 andere programmeurs. 

- [x] Het programma wordt getest en werkt op 2 verschillende browsers (Google Chrome & Edge). 

### Toegankelijkheid 

- [X] De website moet voor iedereen te navigeren zijn. 

- [X] Inclusief mensen met een beperking. 

- [X] De website dient volledig navigeerbaar te zijn zonder gebruik te maken van een muis. 

- [X] Dit betekent navigeren via de tab en enter toets met behulp van spraak. 

- [X] De knoppen zijn voorzien van duidelijke beschrijvingen. 

- [X] De knoppen zijn groot genoeg om de tekst te bevatten. 

- [X] De teksten op de website bevatten geen spelfouten en zijn duidelijk geformuleerd. 

- [X] Zorg voor voldoende contrast. 

- [X] Contrast wordt getest met deze tool: https://webaim.org/resources/contrastchecker/  

- [X] De WCAG AAA test moet pass teruggeven. 

- [X] Maak geen gebruik van hyperlinks met een klik hier maar gebruik een betekenisvolle beschrijving.  

Zoals: Lees meer over regulering. 

- [X] Maak korte en inhoudelijke pagina’s. 

## Technische definities 

### Reviewers 

- [ ] Elke pull request is door minimaal 2 personen nagekeken. 

- [ ] Een taak wordt pas als af beschouwd zodra hij is goedgekeurd door de 2 reviewers en samengevoegd is op de juiste branch. 

- [ ] De reviewer doorloopt de happy flow. 

- [ ] Hierbij maakt de reviewer gebruik van de juiste werkwijze van de website. 

- [ ] De reviewer probeert d.m.v. onjuist gebruik van de website de website te breken. 

- [ ] Dit kan hij/zij doen door verkeerde invoer van velden. 

- [ ] Views zijn getest op responsiveness (mobiel, tablet, desktop) door middel van UI testen. 

### Ontwikkelaar 

- [X] Er is geen SQL injection mogelijk  

- [X] Alle invoervelden zijn voorzien van een regex. 

- [X] Deze regex zorgt ervoor dat alleen het noodzakelijke ingevoerd kan worden. 

- [X] De coding guidlines zijn aangehouden. 

- [X] De geschreven code compileert. 

- [X] De geschreven code veroorzaakt geen crashes. 

- [X] Code levert geen complicaties op met de AVG-wetgeving. 

- [X] Dusk tests worden geschreven voor de happy flow en minimaal 1 bad flow. 

- [X] Views zijn getest op responsiveness (mobiel, tablet, desktop) door middel van UI testen. 

- [X] De siteMap en navigatiebalk zijn geüpdatet op basis van de aanpassingen. 

### Toegankelijkheid 

- [X] Houd je aan de standaarden voor het web. 

- [X] De website moet door gebruik van tab te navigeren zijn. 

- [X] Voeg een alt-tekst toe bij niet-tekstuele-content met betekenisvolle informatie. 

- [X] Maak geen gebruik van spaties. 

- [X] De knoppen hebben een minimaal formaat van 21x30px. 

- [X] Maak gebruik van een rustige layout. 

- [X] Maak gebruik van koppen met h1, h2, h3 tags. 

- [X] Maak gebruik van beschrijvende kopteksten. 

- [X] Duidelijke benaming voor knoppen. 

- [X] Test zelf je code-kwaliteit via www.validator.w3.org  

- [X] Test je syntax. 

- [X] Maak niet alleen gebruik van kleuren om iets duidelijk te maken of uit te leggen maar ook van vormen.  

- [x] Inlaad snelheid zo hoog mogelijk. 

## Summary
Een nieuwspagina waar nieuwsbrieven en nieuwsartikelen zijn te zijn

### Description

- De Nieuwsbrieven en artikelen kunnen worden bekeken, gemaakt, aangepast en verwijderd worden. 
- Daarnaast kan de data gesorteerd worden op titel en datum aflopend en oplopend.
- Er is een sidebar waar snel genavigeerd kan worden naar datums
- Bij de nieuwsbrieven zijn pdf's die vergroot kunnen worden voor de leesbaarheid


### User story 

- User story s8s 39, 44, 40
- 39 Als beheerder wil ik nieuwsartikelen kunnen beheren, zodat iedereen up-to-date kan blijven met wat er gebeurt.
- 44 Als websitebezoeker wil ik het nieuws overzicht pagina kunnen zien, zodat ik weet wat er gaande is binnen de club. Koppelen aan een data-overzicht
- 40 Als beheerder wil ik nieuwsbrieven beheren, zodat iedereen up-to-date kan blijven met wat er gebeurt.

### List of acceptation criteria
39
 o   Nieuws kan toegevoegd worden

o   Nieuws kan verwijderd worden

o   Nieuws kan aangepast worden

o   Bevat:

§  Bestanden (optioneel)

§  Foto’s (optioneel)

§  Datum

§  Tekst

§  Titel
40
   Een nieuwsbrief bestaat uit een pdf-bestand en een datum

o   Nieuwsbrieven kunnen toegevoegd worden

o   Nieuwsbrieven kunnen verwijderd worden

Nieuwsbrieven kunnen aangepast worden

44
Het nieuws kan worden ingezien

o   Er kan gefilterd worden

§  Op datum

§  Op titel

o   Er kan geordend worden

§  Op datum

§  Op titel

## Challenges and Solutions

### Challenge 1

- Het mergen van de nieuwsbrieven, de nieuwsartikelen, en het overzicht ging meerdere keren niet soepel. Dit komt omdat de user stories niet goed independent waren. 

### Solution 1

- Af en toe handmatig wat gemerged. En contact gelegd met de mensen die werkten aan deze user stories.

## User Interface

- ![image](https://github.com/Mees-H/SCRUM-SOp/assets/101800635/0a6129f5-a02b-4ede-b98d-1f2c8c7aaeb9)

EvaVanGurp committed:
Database model

EvaVanGurp committed:
Nieuwsbrief frontend

PerryWdm committed:
Added layout structure newspage

PerryWdm committed:
Adding sidebar to newspage

PerryWdm committed:
Refining sidebar

PerryWdm committed:
Changes sidebar

EvaVanGurp committed:
Nieuws artikelen CRUD deel 1

PerryWdm committed:
Added pdf's in overview page

EvaVanGurp committed:
Nieuwsartikel CRUD deel 2

PerryWdm committed:
Reorganized layout

PerryWdm committed:
Merged overview with articles

EvaVanGurp committed:
Nieuws Tests

EvaVanGurp committed:
Update NewsArticleController.php

PerryWdm committed:
Changed create newspage layout

PerryWdm committed:
Added filter

PerryWdm committed:
Working on date article picker

lauravdh1 committed:
nieuwsbrief maken controller

PerryWdm committed:
Sorting works now

PerryWdm committed:
Changing button and removing duplicate code

PerryWdm committed:
Shows the right data in the sidebar

PerryWdm committed:
Right sorting

PerryWdm committed:
removed unused method

EvaVanGurp committed:
AppServiceProvider fix

lauravdh1 committed:
CRUD nieuwsbrief

lauravdh1 committed:
added create edit

added create edit views
added NewsLetterSeeder
updated nieuwsbrief view

Commit:
fixed responsiveness

Commit:
Revert "Merge branch 'feature/nieuws' of https://github.com/Mees-H/SCRUM-SOp into feature/nieuws"

This reverts commit ff0181fca1baa1dc023d903fda5957283c9dc527, reversing
changes made to 98d1a1aa7b56521ce1718ffedfad634eeec8c70a.

Commit:
fixed responsiveness

PerryWdm committed:
fixing merge problems layout newsarticles

PerryWdm committed:
sidebar fix

PerryWdm committed:
fixed routing

PerryWdm committed:
fixing layout

PerryWdm committed:
fixed adding newsarticle layout

PerryWdm committed:
finished nieuwsoverzicht layout

PerryWdm committed:
accessability fix

PerryWdm committed:
merging newsletters with newsarticles

PerryWdm committed:
Added newsletter to the article page

PerryWdm committed:
addding some files

PerryWdm committed:
finished layout

lauravdh1 committed:
nieuwsbrief tests

PerryWdm committed:
added test

PerryWdm committed:
removed pseudo

PerryWdm committed:
reformatted code

PerryWdm committed:
added alttekst

webwizard678 commented on app/Http/Controllers/NewsArticleController.php:
```diff
@@ -0,0 +1,248 @@
+                $imgArr[] = $image_name;
+            }
+            $article['imgurl'] = $imgArr;
+        }
```
Als het goed is hebben we nu 1 plek waar we afbeeldingen opslaan. kijk even hoe dit gedaan is en doe dit op dezelfde manier.

webwizard678 commented on app/Http/Controllers/NewsArticleController.php:
```diff
@@ -0,0 +1,248 @@
+    public function show(string $id)
+    {
+        //
+    }
```
als dit toch niet gebruikt wordt kan dit gewoon weg

webwizard678 commented on app/Http/Controllers/NewsArticleController.php:
```diff
@@ -0,0 +1,248 @@
+            $newsLetters = Newsletter::all()->sortBy('title');
+        } catch(ModelNotFoundException $e){
+            return redirect('/nieuws')->with('error', 'Kon artikel niet ophalen');
+        }
```
try catch is niet zo mooi, niet fout maar niet zo mooi. je kunt dit anders doen door:
$editArticle = NewsArticle::find($id);
if ($editArticle != null){}

webwizard678 commented on public/storage/img/nieuws/Moon.jpg:
Als je een foto toevoegt aan de repo maak het dan wel een logische foto. Niet een van de maan...

webwizard678 commented on resources/views/nieuws/create.blade.php:
kijk nog eens goed naar deze pagina. het is erg onlogisch geordend. De 2 crud systemen staan onder elkaar met daartussen de al bestaande items. Dit is verwarrend

webwizard678 commented on resources/views/nieuws/create.blade.php:
Run ook even axe devtools er over heen en fix die errors

JoshavEngelen replied:
Ik neem aan niet bij het crud scherm toch? Want dat blijkt niet nodig te zijn

webwizard678 commented on resources/views/nieuws/edit.blade.php:
zelfde feedback als bij de create versie van dit.
[resources/views/nieuws/create.blade.php](https://github.com/Mees-H/SCRUM-SOp/pull/68/files#diff-30b182a82637b962b11fb468bde93e9f43d82bb913f20e4bbc82bf1be66dd3e9)

webwizard678 commented on resources/views/nieuws/nieuwsbrief.blade.php:
ook hier zelfde feedback
[resources/views/nieuws/create.blade.php](https://github.com/Mees-H/SCRUM-SOp/pull/68/files#diff-30b182a82637b962b11fb468bde93e9f43d82bb913f20e4bbc82bf1be66dd3e9)

webwizard678 reviewed (dismissed):
Kijk even naar mijn comments.
Maak ook een paar unit tests.
Pas de verschillende data weergave aan naar nederlands: dd-mm-yyyy

JoshavEngelen commented on app/Http/Controllers/NewsArticleController.php:
```diff
@@ -0,0 +1,248 @@
+    //filter on title desc
+    public function filterTitleDesc($view){
+        $articles = NewsArticle::all()->sortByDesc('title');
+        $newsLetters = Newsletter::all()->sortByDesc('title');
```
Nieuwsbrieven hebben geen titel, toch?

JoshavEngelen commented on app/Http/Controllers/NewsArticleController.php:
```diff
@@ -0,0 +1,248 @@
+    public function filterTitleAsc($view)
+    {
+        $articles = NewsArticle::all()->sortBy('title');
+        $newsLetters = Newsletter::all()->sortBy('title');
```
Hier ook

JoshavEngelen commented on app/Http/Controllers/NewsArticleController.php:
```diff
@@ -0,0 +1,248 @@
+    /**
+     * Remove the specified resource from storage.
+     */
+    public function destroy(string $id): RedirectResponse
```
Als een artikel dezelfde foto gebruikt als een ander artikel en deze vervolgens wordt verwijderd, wordt deze foto ook verwijderd waardoor het andere artikel geen foto meer heeft. Ff checken

JoshavEngelen commented on routes/web.php:
```diff
@@ -73,6 +75,21 @@
+//News routes
+Route::resource('nieuws', NewsArticleController::class);
+Route::resource('nieuwsbrief', NewsLetterController::class);
+//Route::get('/nieuws', [NewsArticleController::class, 'sorting'])->name('sorting');
```
Als deze niet nodig is, kun je 'm weghalen, neem ik aan?

lauravdh1 replied:
ja

lauravdh1 committed:
nieuwsbrief 01-04-2023

PerryWdm committed:
pulled dev into nieuws branch

PerryWdm committed:
code analysis alterations

lauravdh1 committed:
maan en comment weggehaald

PerryWdm committed:
Layout alterations pr review

PerryWdm committed:
chore

lauravdh1 committed:
afbeeldingen locatie veranderd

comments weggehaald
img opslaan locatie veranderd
$newsLetters = Newsletter::all()->sortByDesc('title'); weggehaald

lauravdh1 committed:
Merge branch 'feature/nieuws' of https://github.com/Mees-H/SCRUM-SOp into feature/nieuws

lauravdh1 committed:
golf.jpg

PerryWdm committed:
fixed image removal of more articles

lauravdh1 committed:
datum weergave aangepast

lauravdh1 committed:
Merge branch 'feature/nieuws' of https://github.com/Mees-H/SCRUM-SOp into feature/nieuws

PerryWdm committed:
de tact

PerryWdm committed:
div remove

lauravdh1 committed:
choose file text gefixt

lauravdh1 committed:
Merge branch 'feature/nieuws' of https://github.com/Mees-H/SCRUM-SOp into feature/nieuws

lauravdh1 committed:
feedback verwerkt

tekst upload afbeelding gefixt
plaatjes selecteren bij verwijder foto's
styling artikel aanpassen

lauravdh1 committed:
PDF NIET MEER VERDWIJNEN PLS

lauravdh1 committed:
title aangepast

lauravdh1 committed:
nieuws tests

lauravdh1 committed:
fix date

lauravdh1 committed:
Merge remote-tracking branch 'origin/dev' into feature/nieuws
